services/wfm/src/main/java/org/openkilda/wfm/topology/event/NetworkTopologyBolt.java

```diff
@@ -0,0 +1,198 @@
+import org.openkilda.messaging.info.InfoMessage;
+import org.openkilda.messaging.info.event.IslInfoData;
+import org.openkilda.messaging.info.event.PathNode;
+import org.openkilda.messaging.info.event.PortInfoData;
+import org.openkilda.messaging.info.event.SwitchInfoData;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.persistence.TransactionManager;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+import org.openkilda.wfm.AbstractBolt;
+import org.openkilda.wfm.share.mappers.IslMapper;
+import org.openkilda.wfm.share.mappers.PortMapper;
+import org.openkilda.wfm.share.mappers.SwitchMapper;
+import org.openkilda.wfm.topology.event.service.IslService;
+import org.openkilda.wfm.topology.event.service.PortService;
+import org.openkilda.wfm.topology.event.service.Sender;
+import org.openkilda.wfm.topology.event.service.SwitchService;
+import org.openkilda.wfm.topology.reroute.bolts.RerouteBolt;
+
+import org.apache.storm.kafka.bolt.mapper.FieldNameBasedTupleToKafkaMapper;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.task.TopologyContext;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class NetworkTopologyBolt extends AbstractBolt {
+
+    private static final Logger logger = LoggerFactory.getLogger(RerouteBolt.class);
+
+    public static final String REROUTE_STREAM = "reroute-stream";
+
+    private transient SwitchService switchService;
+    private transient IslService islService;
+    private transient PortService portService;
+
+    private PersistenceManager persistenceManager;
+
+    private int islCostWhenPortDown;
+
+    public NetworkTopologyBolt(PersistenceManager persistenceManager, int islCostWhenPortDown) {
+        this.persistenceManager = persistenceManager;
+        this.islCostWhenPortDown = islCostWhenPortDown;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void prepare(Map stormConf, TopologyContext context, OutputCollector collector) {
+        TransactionManager transactionManager = persistenceManager.getTransactionManager();
+        RepositoryFactory repositoryFactory = persistenceManager.getRepositoryFactory();
+        this.switchService = new SwitchService(transactionManager, repositoryFactory);
+        this.islService = new IslService(transactionManager, repositoryFactory);
+        this.portService = new PortService(transactionManager, repositoryFactory, islCostWhenPortDown);
+        super.prepare(stormConf, context, collector);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void handleInput(Tuple tuple) {
+        String request = tuple.getString(1);
+
+        BaseMessage baseMessage;
+        try {
+            baseMessage = MAPPER.readValue(request, BaseMessage.class);
```

Comment:
could you pass java object to this bolt? if so - you won't need to read it from json again, because we already did it in OfeLinkBolt

Reply:
This is fixed.